packages/react-dom-bindings/src/shared/ReactControlledValuePropTypes.js

```diff
@@ -35,8 +35,8 @@ export function checkControlledValueProps(
       console.error(
         'You provided a `value` prop to a form field without an ' +
           '`onChange` handler. This will render a read-only field. If ' +
-          'the field should be mutable use `defaultValue`. Otherwise, ' +
-          'set either `onChange` or `readOnly`.',
+          'the field should be mutable use `defaultValue`. Otherwise, set %s.',
+        tagName === 'select' ? '`onChange`' : 'either `onChange` or `readOnly`',
       );
```

Comment:
nit:

Can you please split it into `if / else` statement?
```
if (tagName === 'select') {
  console.error('...');
} else {
  console.error('...');
}
```

I think this is much more readable